pandas/tests/frame/methods/test_drop.py

```diff
@@ -172,7 +172,7 @@ def test_drop_multiindex_not_lexsorted(self):
         lexsorted_mi = MultiIndex.from_tuples(
             [("a", ""), ("b1", "c1"), ("b2", "c2")], names=["b", "c"]
         )
-        lexsorted_df = DataFrame([[1, 3, 4]], columns=lexsorted_mi)
```

Comment:
Instead of changing the input, can you change `expected` instead. Changing the input changes our test coverage.

Reply:
Thanks for the feedback!

The new commit removes `tm.assert_frame_equal(lexsorted_df, not_lexsorted_df)` instead considering that 1) pivoting `not_lexsorted_df` now results in columns of floats, and 2) this statement appears to serve as an intermediary assertion? Please let me know if there is a more appropriate way of handling this.

Reply:
This looks good - could you also not drop `a` from each or does the test fail if `a` is included in both?

Reply:
If I'm not mistaken, wouldn't removing `.drop` go against the purpose of this test in particular? This test seems to have been included in PR #11717 to investigate `.drop` between non-lexsorted and lexsorted hierarchical indices as detailed in issue #12078.

To answer the question though, if `a` is included in both `expected` and `result` by removing `.drop`, the test would fail under the resulting committed change of `expected = lexsorted_df.astype(float)`, which would need to be modified to not cast column `a` as float to match the index column of `result`:

```
In [1]:  expected = lexsorted_df.astype(float)
In [2]:  expected
Out[2]:
b    a   b1   b2
c        c1   c2
0  1.0  3.0  4.0

In [3]:  result = not_lexsorted_df
In [4]:  result
Out[4]:
b  a   b1   b2
c      c1   c2
0  1  3.0  4.0
```

Reply:
Indeed; thanks. I missed this was in test_drop.